src/filter/addons/filter-item-checkbox-helpers.ts

```diff
@@ -0,0 +1,137 @@
+import { FilterItemCheckboxOptionProps } from "../types";
+import { FlattenedFilterOption } from "./types";
+
+// =============================================================================
+// HELPER FUNCTIONS FOR NESTED OPTIONS
+// =============================================================================
+
+/**
+ * Builds a key path string from an array of keys
+ */
+export const buildKeyPath = (keyPath: string[]): string => {
+    return keyPath.join(".");
+};
+
+/**
+ * Flattens nested options into a flat array for rendering
+ */
+export const flattenNestedOptions = <T = FilterItemCheckboxOptionProps>(
+    nestedOptions: T[],
+    parentKeyPath: string[] = [],
+    valueExtractor?: (item: T) => string,
+    labelExtractor?: (item: T) => string
+): FlattenedFilterOption<T>[] => {
+    const flattened: FlattenedFilterOption<T>[] = [];
+
+    nestedOptions.forEach((option, index) => {
+        const value = valueExtractor
+            ? valueExtractor(option)
+            : (option as any).value;
+        const label = labelExtractor
+            ? labelExtractor(option)
+            : (option as any).label;
+        const options = (option as any).options;
+        const keyPath = [...parentKeyPath, String(value)];
+        const hasChildren = !!(options && options.length > 0);
+
+        flattened.push({
+            originalItem: option,
+            value,
+            label,
+            keyPath,
+            level: parentKeyPath.length,
+            hasChildren,
+            parentKeyPath: parentKeyPath.length > 0 ? parentKeyPath : undefined,
+            indexInParent: index,
+            parentSetSize: nestedOptions.length,
+        });
+
+        // Recursively flatten children
+        if (hasChildren) {
+            flattened.push(
+                ...flattenNestedOptions(
+                    options,
+                    keyPath,
+                    valueExtractor,
+                    labelExtractor
+                )
+            );
+        }
+    });
+
+    return flattened;
+};
+
+/**
+ * Gets all descendant key paths (children, grandchildren, etc.) for a given parent
+ */
+export const getAllDescendantKeyPaths = <T = FilterItemCheckboxOptionProps>(
+    flattenedOptions: FlattenedFilterOption<T>[],
+    parentKeyPath: string[]
+): string[][] => {
+    const parentKeyPathString = buildKeyPath(parentKeyPath);
+
+    return flattenedOptions
+        .filter((option) => {
+            const optionKeyPathString = buildKeyPath(option.keyPath);
+            return (
+                option.keyPath.length > parentKeyPath.length &&
+                optionKeyPathString.startsWith(parentKeyPathString + ".")
```

Comment:
can pull this check into a `isChild(parentKeyPath, childKeyPath)` helper since it's used in both getAllDescendantKeyPaths and getAllDescendantChildren